cdist/conf/explorer/init

```diff
@@ -0,0 +1,34 @@
+#!/bin/sh
+#
+# 2016 Daniel Heule (hda at sfs.biz)
+#
+# This file is part of cdist.
+#
+# cdist is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# cdist is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with cdist. If not, see <http://www.gnu.org/licenses/>.
+#
+#
+# Returns the process name of pid 1 ( normaly the init system )
+# for example at linux this value is "init" or "systemd" in most cases
+#
+
+uname_s="$(uname -s)"
+
+case "$uname_s" in
+    Linux|FreeBSD)
```

Comment:
Hello Nico,

i have studied the man page, modified the --pid to -p, all other options to
ps should work as stated in the man page.
But: I don't have a FreeBSD to test it ...



2016-03-24 15:57 GMT+01:00 Nico Schottelius <notifications@github.com>:

> In cdist/conf/explorer/init
> <https://github.com/ungleich/cdist/pull/421#discussion_r57329325>:
>
> > +# but WITHOUT ANY WARRANTY; without even the implied warranty of
> > +# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
> > +# GNU General Public License for more details.
> > +#
> > +# You should have received a copy of the GNU General Public License
> > +# along with cdist. If not, see <http://www.gnu.org/licenses/>.
> > +#
> > +#
> > +# Returns the process name of pid 1 ( normaly the init system )
> > +# for example at linux this value is "init" or "systemd" in most cases
> > +#
> > +
> > +uname_s="$(uname -s)"
> > +
> > +case "$uname_s" in
> > +    Linux|FreeBSD)
>
> Oh - just seen this line - have you tested that this ps works on FreeBSD?
> I thought it doesn't!
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/ungleich/cdist/pull/421/files/9c6d7dbd892c30c11ed8c03f74a24d688d47cad4#r57329325>
>